### PR TITLE
[ELY-440] Permissions framework

### PR DESCRIPTION
**Pull request by dmlloyd**

### CODE REVIEW

File: src/main/java/org/wildfly/security/permission/AbstractActionSetPermission.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    public final This withActionBits(int actionBits) {
+        return withNewActionBits(this.actionBits | actionBits & actionsMask());
+    }
+
+    public final This withoutActions(String actionsString) {
+        return withoutActionBits(parseActions(actionsString));
+    }
+
+    public final This withoutActionsFrom(final This permission) {
+        return withActionBits(permission.getActionBits());
```
Comment (fjuma) [inline src/main/java/org/wildfly/security/permission/AbstractActionSetPermission.java]: Looks like this should be withoutActionBits(permission.getActionBits()).

Comment (dmlloyd) [inline src/main/java/org/wildfly/security/permission/AbstractActionSetPermission.java]: Fixed.

**Comment by dmlloyd** [comment]
Rebased

**Comment by dmlloyd** [comment]
Incorporated the new CredentialStorePermission class.

**Pull request merged**